### PR TITLE
fix(elasticloadbalancingv2): should allow more than 2 certificates

### LINKED ISSUE #13150: (elasticloadbalancingv2): adding a new certificate to a multi-certificate listener will lose old extra ceritifactes
If you try to add a new certificate to a listener that has already at least one extra certificate (i.e. one in addition to the main certificate), the old extract certificate will be lost by the end of update operation.

### Reproduction Steps

(In my case, it happened like this:)
1. Create an HTTPS listener with two certificates; `foo.com` (main one) and `bar.com` (additional one) stack and deploy it.
2. Change the code and add a new certificate; `baz.com` to the list and deploy it.

### What did you expect to happen?

The listener should keep one main certificate and two extra certificates.

### What actually happened?

The old extra certificate (`bar.com`) is gone.

### Environment

  - **CLI Version      :** 1.89.0
  - **Framework Version:** 1.89.0 (build df7253c)
  - **Node.js Version:** v15.5.1
  - **OS               :** Mac OS X 11.2
  - **Language (Version):** typescript ~3.8.2

### Other

Please note that, according to my findings from AWS support, this seems to be a bug/feature of CloudFormation. Interestingly, even though the `Certificates` field is of type `Array`, the documentation says: _You can specify one certificate per resource._
If my understanding is correct, CDK should not put all additional certificates into one `AWS::ElasticLoadBalancingV2::ListenerCertificate` resource. It should be one `AWS::ElasticLoadBalancingV2::ListenerCertificate` resource per each additional certificate.

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by andreialecu**
Fixes #13150 

> Interestingly, even though the Certificates field is of type Array, the documentation says: You can specify one certificate per resource.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by andreialecu** [comment]
Not sure why CI is failing, but it doesn't seem related to the changes in this PR.

```
Checking if ruby is installed... which: no ruby in (/codebuild/output/src536429546/src/github.com/aws/aws-cdk/node_modules/.bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/codebuild/user/bin)
Missing dependency: ruby. Install ruby >= 2.5.1
```

File: packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts [hunk 1]
```diff
@@ -219,10 +219,12 @@ export class ApplicationListener extends BaseListener implements IApplicationLis
       this.certificateArns.push(first.certificateArn);
     }
 
-    if (additionalCerts.length > 0) {
+    // Only one certificate can be specified per resource, even though 
+    // `certificates` is of type Array
+    for (const cert of additionalCerts) {
       new ApplicationListenerCertificate(this, id, {
```
Comment (njlynch) [inline packages/@aws-cdk/aws-elasticloadbalancingv2/lib/alb/application-listener.ts:225]: The `id` parameter will need to change; we can't have multiple constructs with the same id in the same scope. Should be as simple as appending a counter to the id, although for backwards-compatibility reasons, we would probably only want to append the id to the 2+ object. (e.g., 'id', 'id2', 'id3').

**Review by njlynch — DISMISSED** [review]
Thanks for the contribution!

In addition to the below comment about IDs, can you add a test for this? Ideally we'd add 3 certs, and verify both `ApplicationListenerCertificate` were created.

**Comment by njlynch** [comment]
> Not sure why CI is failing, but it doesn't seem related to the changes in this PR.

Agreed; looks like either a general breakage or a solar-flare-induced hiccup. I'd say you can safely ignore it. You can run the tests just for this module by running `lt` in the module's directory. (See https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md#useful-aliases for the alias setup).

**Comment by andreialecu** [comment]
Note that I have also received a response from AWS support regarding this behavior. The support engineer mentioned:

> During my research into this, I found that this is a known issue that AWS CloudFormation service team are aware of this behavior. You are totally correct about your statement.

> I checked that an internal ticket to the team is currently being investigated and I have added your voice onto the ticket to make the team aware that you are also facing this behavior.

> => I would suggest you to kindly use multiple separate "AWS::ElasticLoadBalancingV2::ListenerCertificate" resources with only one certificate per resource. This will avoid the need to update the resource "ListenerCertificates" which currently removes the old certificates during the update process.

**Comment by andreialecu** [comment]
@njlynch I added the tests. Let me know if any further changes are required.

**Review by njlynch — APPROVED** [review]
Thanks for the fix!

**Pull request merged**